crates/autopilot/src/risk_adjusted_rewards.rs

```diff
@@ -181,6 +203,91 @@ fn uncapped_reward_eth_atoms(
     (profit + cost) / (1. - revert_probability) - cost
 }
 
+/// Price cache for [BestEffortCowPriceEstimator].
+struct FallbackCache {
+    /// Cached value and a timestamp.
+    value: Option<(f64, Instant)>,
+    /// How long is the value valid after being set.
+    max_age: Duration,
+}
+
+impl FallbackCache {
+    fn get(&mut self, current_time: Instant) -> Option<f64> {
+        let (price, timestamp) = self.value?;
+
+        if timestamp + self.max_age >= current_time {
+            Some(price)
+        } else {
+            None
+        }
+    }
+
+    fn set(&mut self, price: f64, current_time: Instant) {
+        self.value.replace((price, current_time));
+    }
+}
+
+/// A caching wrapper over [NativePriceEstimating] used to estimate and cache
+/// CoW token price. Implemented to enable the [Calculator] produce rewards even
+/// when a fresh estimate is not available.
+#[derive(Clone)]
+struct BestEffortCowPriceEstimator {
+    inner: Arc<dyn NativePriceEstimating>,
+    cow_token: H160,
+    fallback_cache: Arc<Mutex<FallbackCache>>,
```

Comment:
This doesn't need Arc. Also the mutex is only needed for `value: Option<(f64, Instant)>`.

(less sure about the following) I would consider inlining `max_age` and `value` into this struct. You can still have the `get` and `set` methods for the unit test.

Reply:
Indeed, thanks. I flattened the estimator and removed `FallbackCache`